File: src/main/java/com/rimlang/rim/lexer/GroupToken.java
package com.rimlang.rim.lexer;

import java.util.List;

public class GroupToken extends Token {
    private final List<Token> value;

    public GroupToken(TokenType type, List<Token> value, int line, int character) {
        super(type, line, character);
        this.value = value;
    }

    public List<Token> getValue() {
        return value;
    }

    @Override
    public Object value() {
        return value;
    }
}

File: src/main/java/com/rimlang/rim/lexer/token/GroupToken.kt
package com.rimlang.rim.lexer.token

import com.rimlang.rim.errors.CodeFile

class GroupToken(
    type: TokenType,
    override val value: List<Token>,
    line: Int,
    character: Int,
    file: CodeFile
) : Token(type, line, character, file)